source/tagged-union.d.ts

```diff
@@ -0,0 +1,41 @@
+/**
+Utility type to create a tagged union with given tag
+
+Use-case: A shorter way to declare tagged unions with multiple members.
+
+@example
+```
+import type {TaggedUnion} from 'type-fest';
+
+type EventMessage = TaggedUnion<
```

Comment:
I think it would be useful to show the same an example of the same just without this type. So you can clearly see how it simplifies making a tagged union.

Reply:
True! I added an example without use of this utility.